pkg/kubelet/kubelet.go

```diff
@@ -1343,9 +1322,7 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	go kl.volumeManager.Run(kl.sourcesReady, wait.NeverStop)
 
 	if kl.kubeClient != nil {
-		// Start syncing node status immediately, this may set up things the runtime needs to run.
-		go wait.Until(kl.syncNodeStatus, kl.nodeStatusUpdateFrequency, wait.NeverStop)
-		go kl.fastStatusUpdateOnce()
+		kl.nodeStatusManager.Start()
```

Comment:
this should be in a separate goroutine.  It actually looks like the common pattern is a function called `Run(stopCh chan)`.  I'm guessing that will make unit testing easier.

Reply:
It's better, but I think the really tricky thing is nodeStatus unit tests all depends on kubelet struct. I'll try to handle it.